experimental/auth-logic/wrappers/rekor_wrapper.go

```diff
@@ -122,18 +152,107 @@ func verifyRekordLogSignature(rekordEntry *rekord.V001Entry) (*ecdsa.PublicKey,
+		return fmt.Errorf("Input product team key does not match rekor log entry key: %v, %v", logECDSAPubKey, prodTeamECDSAPubKey)
 	}
+	return nil
+}
 
-	sig := rekordEntry.RekordObj.Signature.Content
+// EmitStatement returns the unattributed statement for the rekor log wrapper
+func (rlw RekorLogWrapper) EmitStatement() (UnattributedStatement, error) {
+	// Get principal names for the endorsement file and rekor log entry
+	// by using the app name from the endorsement file
+	endorsementAppName, err := GetAppNameFromEndorsement(rlw.endorsementFilePath)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("could not get app name from endorsement file: %s, %v", rlw.endorsementFilePath, err)
+	}
+	endorsementPrincipal := fmt.Sprintf(`"%s::EndorsementFile"`, SanitizeName(endorsementAppName))
+	logEntryPrincipal := fmt.Sprintf(`"%s::RekorLogEntry"`, SanitizeName(endorsementAppName))
 
-	ok = ecdsa.VerifyASN1(ecdsaKey, data, sig)
-	if !ok {
-		return nil, fmt.Errorf("could not verify ecdsa signature. key:%v, data:%v, sig:%v ", ecdsaKey, data, sig)
+	// Unpack rekord log entry from bytes into go structs
+	logEntryAnon, err := getLogEntryAnonFromBytes(rlw.rekorLogEntryBytes)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't parse rekor log entry from bytes: %v, %v", rlw.rekorLogEntryBytes, err)
+	}
+	entryImpl, err := getEntryImplFromAnon(*logEntryAnon)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't get entryImpl from body of logEntryAnon %v: %v", *logEntryAnon, err)
 	}
 
-	return ecdsaKey, nil
+	rekordEntry, err := getRekordEntryFromEntryImpl(*entryImpl)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't get rekordEntry from entryImpl:, %v %v", *entryImpl, err)
+	}
+
+	// Authorization logic statmenets are defined next to the go code that
+	// they correspond to so that this is easier to follow along
+
+	// Verify rekor log entry signature
+	_, err = verifyRekordLogSignature(rekordEntry)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't validate signature in rekor log entry %v", err)
+	}
+	logEntrySignatureStatement := fmt.Sprintf("hasValidBodySignature(%v).", logEntryPrincipal)
+
+	// Verify inclusion proof
+	err = checkInclusionProof(logEntryAnon, strfmt.Default)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("couldn't validate logEntryAnon (which includes inclusion proof checking):%v ", err)
+	}
+	inclusionProofStatement := fmt.Sprintf("hasValidInclusionProof(%v).", logEntryPrincipal)
+
+	// Check that the product team public key in the log entry matches
+	// the input public key
+	err = checkEntryPubKeyMatchesExpectedKey(rekordEntry, rlw.productTeamKeyBytes)
+	if err != nil {
+		return UnattributedStatement{}, fmt.Errorf("rekord entry key does not match input product team key: %v, %v, %v", rekordEntry, rlw.productTeamKeyBytes, err)
+	}
+	pubKeyMatchStatement := fmt.Sprintf("hasCorrectPubkey(%v).", logEntryPrincipal)
```

Comment:
I guess what we care most about is the identity of the signer. Is not it?

Reply:
I think so too... Is this comment meant to request a change to the code? I am not sure what change that is if so :)

This auth logic statement is meant to correspond to the check that the pubkey in the rekord entry (which we also use for verifying the signature in the entry) is the same as another input pubkey for the product team that we trust. This check is the same as checking for "the correct identity of the signer". Was your comment a request to phrase this differently from "hasCorrectPubkey" ?

Reply:
My comment is aimed at clarifying the purpose of this statement in the auth logic code :)
 
I'd suggest rephrasing it to be clear that here we are checking the identity of the signer, if it makes sense from the auth logic policy perspective.